server/src/test/java/org/elasticsearch/search/aggregations/metrics/percentiles/InternalPercentilesRanksTestCase.java

```diff
@@ -39,4 +49,52 @@ protected final void assertFromXContent(T aggregation, ParsedAggregation parsedA
         Class<? extends ParsedPercentiles> parsedClass = implementationClass();
         assertTrue(parsedClass != null && parsedClass.isInstance(parsedAggregation));
     }
+
+    public void testEmptyRanksXContent() throws IOException {
```

Comment:
You mentioned earlier you cannot push this test up into AbstractPercentilesTestCase because of some subtle difference, but I cannot spot it. Do you remember what it was? Otherwise I'd give it another try to push it up.

Reply:
It's super tiny:  Percentiles uses `percent() / percentAsString()` while PercentileRanks uses `percentile() / percentileAsString()`.

I could collapse them into a single test and then do an `instanceOf` or `getType()` and switch on that if you think it'd be cleaner.  Less test code duplication, but a bit more fragile.

Reply:
Ah, I see it now. What about pulling the test up and just doing the two lines of assertions that are different in their own little helper method that you overwrite differently in both cases? I'm usually also not a fan of doing so much code acrobatics in tests but in this case I think the gain in non-duplicated lines of code would justify it. I don't think its super important though, thanks for pointing out the difference.

Reply:
++ this cleaned up nicely.  Thanks for the suggestion!